### DISCUSSION TITLE
Server-side-only dependencies?

### DISCUSSION THREAD

**Discussion by steveruizok** [vercel/next.js#24426] [Help]
Hi,

I recently added a dependency to my Next.js project (`playwright-aws-lambda`) that I'm only using on an api route, to generate dynamic social images for certain pages. I noticed that the dependency had an immediate effect on my bundle size (bringing First Load JS from 98kb to 142kb). 

A visitor would not need this code—it will only run on the server—so my question is, does Next provide a way to exclude this dependency from the bundle?

**Comment by gomflo** [comment]
Thats weird https://nextjs.org/docs/basic-features/data-fetching

![image](https://user-images.githubusercontent.com/141670/116112894-9f612a00-a67d-11eb-84f3-9a8705d1518d.png)

https://nextjs.org/docs/basic-features/data-fetching#write-server-side-code-directly

![image](https://user-images.githubusercontent.com/141670/116112998-b738ae00-a67d-11eb-8aa6-11ffcae5b73c.png)

check out this tool: https://next-code-elimination.vercel.app/
